router/http.go

```diff
@@ -47,13 +47,14 @@ type HTTPListener struct {
 	tlsListener net.Listener
 	closed      bool
 	cookieKey   *[32]byte
+	keypair     tls.Certificate
 }
 
 type DiscoverdClient interface {
 	NewServiceSet(string) (discoverd.ServiceSet, error)
 }
 
-func NewHTTPListener(addr, tlsAddr string, cookieKey *[32]byte, ds DataStore, discoverdc DiscoverdClient) *HTTPListener {
+func NewHTTPListener(addr, tlsAddr string, cookieKey *[32]byte, keypair tls.Certificate, ds DataStore, discoverdc DiscoverdClient) *HTTPListener {
```

Comment:
This function is getting pretty overloaded. What do you think about moving to setting fields on the struct directly and then do a lazy init in `Start`?